session/session.go

```diff
@@ -144,6 +144,7 @@ type Session interface {
 	Close()
 	Auth(user *auth.UserIdentity, auth []byte, salt []byte) bool
```

Comment:
Please add some test code for this change.
I understand that unit test may not be a perfect match for this scenario, at least you can  `insert .. into mysql.user` to mock some data, and cover the auth utility?

Reply:
I've added `TestAuthPluginForUser` for this.